src/lib.rs

```diff
@@ -143,19 +143,18 @@ pub trait MemorySubAllocator<B: Backend> {
 /// Calculate shift from specified offset required to satisfy alignment.
 pub fn alignment_shift<T>(alignment: T, offset: T) -> T
 where
-    T: From<u8> + Add<Output = T> + Sub<Output = T> + Rem<Output = T> + Eq + Copy,
+    T: From<u8> + Add<Output=T> + Sub<Output=T> + BitOr<Output=T> + PartialOrd + Copy,
 {
-    if offset == 0.into() {
-        0.into()
-    } else {
-        alignment - (offset - 1.into()) % alignment - 1.into()
-    }
+    shift_for_alignment(alignment, offset) - offset
 }
 
-/// Shift from specified offset in order to to satisfy alignment.
-pub fn shift_for_alignment<T>(alignment: T, offset: T) -> T
+fn shift_for_alignment<T>(offset: T, alignment: T) -> T
 where
-    T: From<u8> + Add<Output = T> + Sub<Output = T> + Rem<Output = T> + Eq + Copy,
+    T: From<u8> + Add<Output=T> + Sub<Output=T> + BitOr<Output=T> + PartialOrd,
 {
-    offset + alignment_shift(alignment, offset)
+    if offset > 0.into() && alignment > 0.into() {
```

Comment:
Bigger than `0`? When would that happen?

Reply:
Offset or alignment?
First one - commonly, second one - almost always.

Reply:
My question was what meaning it would have, it doesn't make any sense to me.